api/models/spell.py

```diff
@@ -0,0 +1,229 @@
+        if "range" in json:
+            # If the spell range includes an effect description like "(10 ft
+            # cube)", remove.
+            range_trimmed = str(json['range'].split(" (")[0]).lower()
+
+            # Set target range sort based on the following rules:
+            # Self = 0
+            # Touch = 1
+            # Any other distance is recorded in feet.
+            # "Sight" is recorded as 9999
+            # Unlimited (same-plane) is recorded as 99990
+            # Unlimited (multi-plane) is recorded as 99999
+
+            if 'self' in range_trimmed:
+                # Various spells in deep magic include effect shape.
+                self.target_range_sort = 0
+            if range_trimmed == 'touch':
+                self.target_range_sort = 1
+
+            if range_trimmed == 'sight':
+                self.target_range_sort = 9999
+
+            if range_trimmed == 'unlimited':
+                self.target_range_sort = 99999
+
+            if range_trimmed == 'special':
+                # For now, assume it's single plane
+                self.target_range_sort = 99990
+
+            if range_trimmed.endswith(
+                    'miles') or range_trimmed.endswith('mile'):
+                # Assume it's in the format "xx Miles"
+                miles = int(json['range'].split(" ")[0])
+                self.target_range_sort = miles * 5280
+
+            if range_trimmed.endswith('feet') or range_trimmed.endswith('ft.'):
+                # Assume it's in the format "xx Miles"
+                feet = int(range_trimmed.split(" ")[0])
+                self.target_range_sort = feet
+
+            self.range = json["range"]
+        if "components" in json:
+            # Set defaults to False, and then set to true based on string of
+            # "V, S, M"
+            self.requires_verbal_components = False
+            self.requires_somatic_components = False
+            self.requires_material_components = False
+            if 'v' in json['components'].lower():
+                self.requires_verbal_components = True
+            if 's' in json['components'].lower():
+                self.requires_somatic_components = True
+            if 'm' in json['components'].lower():
+                self.requires_material_components = True
+
+        if "material" in json:
+            self.material = json["material"]
+
+        # Logic to set boolean based on v1 file import spec (a string).
+        self.can_be_cast_as_ritual = False
+        if "ritual" in json:
+            # Default is false
+            if str(json["ritual"]).lower() == 'yes':
+                self.can_be_cast_as_ritual = True
+
+            if str(json["ritual"]).lower() == 'no':
+                pass  # Already set as False through default.
+
+        if "duration" in json:
+            self.duration = json["duration"]
+
+        # Logic to set a boolean based on v1 file import spec (a string).
+        self.requires_concentration = True
```

Comment:
Typically False is the default value for bool fields. I recommend sticking with that convention.

Reply:
Changed.